src/tools/imgtool/imgtool.cpp

```diff
@@ -22,15 +22,17 @@
     TYPE DEFINITIONS
 ***************************************************************************/
 
-struct imgtool_image
+class imgtool::image
```

Comment:
Hang on, this can't be right - you need to put it inside a `namespace imgtool` block.